analytics.php

```diff
@@ -0,0 +1,231 @@
+
+$id = required_param('id', PARAM_RAW);
+$bn = optional_param('bn', 0, PARAM_INT);
+$group = optional_param('group', 0, PARAM_INT);
+
+$viewinstance = bigbluebuttonbn_view_validator(null, $bn); // In locallib.
+if (!$viewinstance) {
+    print_error('view_error_url_missing_parameters', plugin::COMPONENT);
+}
+
+$meetinginfo = bigbluebuttonbn_get_meeting_info_array($id);
+
+$cm = $viewinstance['cm'];
+$course = $viewinstance['course'];
+$bigbluebuttonbn = $viewinstance['bigbluebuttonbn'];
+
+require_login($course, true, $cm);
+
+require_once("$CFG->libdir/tablelib.php");
+require_once($CFG->dirroot . '/mod/bigbluebuttonbn/lib.php');
+
+$context = context_system::instance();
+
+// Print the page header.
+$pageurl = new moodle_url('/mod/bigbluebuttonbn/analytics.php', ['id' => $id, 'bn' => $bn]);
+$PAGE->set_url($pageurl);
+$PAGE->set_title($bigbluebuttonbn->name);
+$PAGE->set_cacheable(false);
+$PAGE->set_heading($course->fullname);
+
+$download = optional_param('download', '', PARAM_ALPHA);
+
+$table = new \mod_bigbluebuttonbn\analytics\analytics_per_session_table('uniqueid', $PAGE->url);
+$table->is_downloading($download);
+
+if (!$table->is_downloading()) {
+    // Only print headers if not asked to download data.
+    $PAGE->set_title($bigbluebuttonbn->name);
+    $PAGE->set_cacheable(false);
+    $PAGE->set_heading($course->fullname);
+    $PAGE->navbar->add(get_string('view_section_title_analytics', 'bigbluebuttonbn'), $pageurl);
+    echo $OUTPUT->header();
+}
+$wheresql = "meta is not null and bigbluebuttonbnid = ? and log = ? and recordid = ? and userid is not null";
+$table->set_sql('id, meta, userid', "{bigbluebuttonbn_logs}", $wheresql, [
+    $bigbluebuttonbn->id,
+    BIGBLUEBUTTON_LOG_EVENT_SUMMARY,
+    $id
+]);
+$table->define_baseurl($pageurl);
+$meetingsummary = $DB->get_record('bigbluebuttonbn_logs', [
+    'userid' => null,
+    'recordid' => $id,
+    'log' => BIGBLUEBUTTON_LOG_EVENT_SUMMARY
+]);
+$output = "";
+
+// Prepare attendee data for Attention Box.
+$table->setup();
+$table->query_db(0); // No limit, fetch all rows.
+$table->pageable(false);
+$table->close_recordset();
+$attendees = $table->rawdata;
+
+if (isset($meetingsummary->meta)) {
+    $meetingsummary->meta = json_decode($meetingsummary->meta);
+}
+
+if (!$table->is_downloading()) {
+    $tables = "";
+    if (isset($meetingsummary->meta)) {
+        $output .= $OUTPUT->heading($meetingsummary->meta->meetingname, 3);
+        $output .= $OUTPUT->heading(
+            userdate(
+                bigbluebuttonbn_datetime_to_timestamp($meetingsummary->meta->start),
+                get_string('strftimedatetime', 'langconfig')
+            ),
+            6
+        );
+
+        // Overview Box.
+        $overviewtable = new html_table();
+        $overviewtable->head = [
+            get_string('view_analytics_heading_overview', 'bigbluebuttonbn'),
+            '',
+        ];
+        $overviewtable->attributes['class'] = 'w-auto admintable generaltable mr-4';
```

Comment:
Can you also add class `table-bordered`, and perhaps consider making them all the same width, or merging the tables into one?

Reply:
I've considered making the tables e.g. col-md-4 and 6, but @brendanheywood suggested to use 'w-auto' instead. I have not considered merging them all into one table. How did you envision it layed out as one table?

Reply:
w-auto is probaby fine,it just felt a tiny bit cramped on the files table, so perhaps a w-auto + min-width?